### ISSUE TITLE
[Feature suggestion] More intuitive conflict outputs, and ignoring desktop info files

### ISSUE THREAD

**Issue by Roadhog360** [Raytwo/ARCropolis#257]
### Have you read the disclaimer before proceeding?

- [X] I have read the disclaimer.

### What would the feature do?

I'm not sure why the output has to be a .json, but the output is pretty bad and messy, (seemingly) random. I missed something because it was buried in the _absolute pile_ that about four or five mods were conflicting, a few characters and stages. Since these mods share so many files of course this bloated the conflicts file to a huge and random assortment of each of those individual files conflicting. The one I was looking for was buried among them and as far as I could tell there was no rhyme or reason to any of it. My first suggestion is to just get rid of json and use plaintext. Why does it need to be json? This isn't a config file and unless I missed something the game doesn't actually need to read it. Correct me if I am wrong. I also did clean up the other conflicts after. But I was only able to find the actual conflict after sifting through [this huge file](https://cdn.discordapp.com/attachments/228184944407609344/959898000023515207/conflicts.json). While the mods listed seem to always be in the same order, I cannot figure out what order the files themselves are being placed in, in this list. The ONE conflict I was looking for was chara_7. I somehow missed it. I then searched for "aero" but the search function had the case sensitive mode on and I forgot to turn it off, leading my search to fail and me thinking the Aero CSS was not loading for seemingly no reason.

### How should it be implemented?

Again, if there's no need for the file to be json and have this kind of formatting, I believe it could be replaced with plaintext. 

Before I start I'd also like to get to the "desktop info files" part of this request, and ask that files with names such as `.DS_Store` or `desktop.ini` be ignored from the checks. In fact, perhaps ARCropolis could offer to delete them if detected?

Here's how I was thinking. Instead of listing the file, and then the mods it's conflicting with, we should go the other way around. We list the mods that have a conflict, and then the files they share. This probably doesn't look so good, but the general idea is to just make the order of this feel less random. Also, each mod is listed over and over again for every individual file. While I think there's no way to create a solution where something isn't listed multiple times, we could have a solution that lists less things in a duplicate

So let's say we have mod1, mod2 and mod3.

Mod1: file1, file2, file3,
Mod2: file2, file3
Mod3: file2

The output would look like this.This probably doesn't look so good, but the general idea is to just make the order of this feel less random.

So let's say we have mod1, mod2 and mod3.

Mod1: file1, file2, file3
Mod2: file2, file3
Mod3: file1, file2

The output would look like this. 

```
sd:/ultimate/mods/Mod1, sd:/ultimate/mods/Mod2, sd:/ultimate/mods/Mod3:
    file2
sd:/ultimate/mods/Mod1, sd:/ultimate/mods/Mod2:
    file3
sd:/ultimate/mods/Mod1, sd:/ultimate/mods/Mod3:
    file1
```
So we create one entry for all 3 mods because they all share file2. Even though Mod1 and Mod2 also share file3 that gets put in a different entry because mod3 doesn't have it. Mod1 and Mod3 share file1, but that also gets a different entry because Mod2 doesn't have it.

If you'd like further examples of this I'll take the file I gave you and completely convert it to my own format by hand to see how it looks.

**Comment by Raytwo** [comment]
This topic has been discussed for a bit with other people, and while I can see that it could use some work, a more thorough rework of this according to my vision for it would take a long time.

What I can potentially offer to make it more approachable is going from a JSON file to a YAML file (check it out, it's probably more readable for this purpose), rename the extension to .txt (makes it more obvious that it can be opened with a text editor), and eventually (I'm not too OK with that but who knows) display it on the screen.

Does any of this sound better to you until then?

**Comment by Roadhog360** [comment]
Yes. Would it be possible to add some sort of grouping for the current file trees too? At the moment it seems completely random. Entries from the same folder seem completely mixed with entries from mods that modify completely different folders.
